### DISCUSSION TITLE
Rule does not work as expected - Pool Solar Control

### DISCUSSION THREAD

**Discussion by Pace17881** [arendst/Tasmota#19093] [Q&A] [answered]
### PROBLEM DESCRIPTION
My Rule:
`RULE1
ON DS18B20-1#temperature>DS18B20-2#temperature DO Power1 0 ENDON
ON DS18B20-2#temperature>DS18B20-1#temperature DO Power1 1 ENDON`

My temperature readings:
`DS18B20-1 Temperature	29.7 °C
DS18B20-2 Temperature	27.8 °C`

For this case the Power1 should be 0 = OFF. But it is always 1 = ON

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): NodeMCU ESP 8266
- [ ] Tasmota binary firmware version number used: 13.0.0.2
  - [ X ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: Tasmotizer
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
`19:35:52.713 MQT: stat/Solar/RESULT = {"Module":{"18":"Generic"}}
19:35:52.919 MQT: stat/Solar/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"224":"Relay1"},"GPIO5":{"1312":"DS18x201"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}`


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
`19:37:49.802 MQT: stat/Solar/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":129,"Free":382,"Rules":"ON DS18B20-1#temperature>DS18B20-2#temperature DO Power1 0 ENDON ON DS18B20-2#temperature>DS18B20-1#temperature DO Power1 1 ENDON"}}
19:37:50.026 MQT: stat/Solar/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}19:37:50.294 MQT: stat/Solar/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}`


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
`19:38:55.855 CMD: Status 0
19:38:55.862 MQT: stat/Solar/STATUS = {"Status":{"Module":18,"DeviceName":"Solar","FriendlyName":["Solar"],"Topic":"Solar","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
19:38:55.879 MQT: stat/Solar/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:16:49","StartupUTC":"2023-07-10T18:22:06","Sleep":50,"CfgHolder":4617,"BootCount":5,"BCResetTime":"2023-07-10T19:17:14","SaveCount":21,"SaveAddress":"F7000"}}
19:38:55.893 MQT: stat/Solar/STATUS2 = {"StatusFWR":{"Version":"13.0.0.2(tasmota)","BuildDateTime":"2023-07-10T16:16:37","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"356/699"}}
19:38:55.907 MQT: stat/Solar/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Internet",""],"TelePeriod":10,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
19:38:55.928 MQT: stat/Solar/STATUS4 = {"StatusMEM":{"ProgramSize":634,"Free":368,"Heap":21,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62,68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
19:38:55.948 MQT: stat/Solar/STATUS5 = {"StatusNET":{"Hostname":"Solar","IPAddress":"192.168.171.223","Gateway":"192.168.171.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.171.128","DNSServer2":"0.0.0.0","Mac":"AC:0B:FB:D9:F8:5A","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}`

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
If temperature 1 higher than 2 the switch should be OFF. If temperature 1 is lower than 2 the switch should be ON.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
You have no issue in Tasmota, you just have a syntax error, as this is outside of the trigger syntax:
> ON DS18B20-1#temperature>DS18B20-2#temperature

You could have an extra rule saving DS18B20-2#Temperature in a var when updated, and then use that var in your existing rules.
